test/deployframework/context.go

```diff
@@ -71,13 +73,18 @@ func (df *DeployFramework) NewDeployerCtx(
 
 	// validate the reporting-operator image is non-empty when overrided
 	if df.ReportingOperatorImageRepo != "" || df.ReportingOperatorImageTag != "" {
-		err := validateImageConfig(*cfg.MeteringConfig.Spec.ReportingOperator.Spec.Image)
+		reportingOperatorImageConfig := &metering.ImageConfig{
+			Repository: df.ReportingOperatorImageRepo,
+			Tag:        df.ReportingOperatorImageTag,
+		}
+
+		err := validateImageConfig(*reportingOperatorImageConfig)
```

Comment:
@chancez I think it may make more sense to build up an `metering.ImageConfig` object with the deployframework reporting-operator repo/tag fields, and validating that object instead of directly validating the `cfg.MeteringConfig...ReportingOperator.Spec.Image` object and having to deal with null checking nested structure fields (like we do with the metering-operator image).

In this case, we can error if the overridden image was improperly managed, which means that metering won't be installed.